src/main/kotlin/dev/stalla/parser/namespace/AtomParser.kt

```diff
@@ -39,18 +39,12 @@ internal object AtomParser : NamespaceParser() {
         atomBuilder: AtomBuilder
     ) {
         when (localName) {
-            "contributor" -> {
-                val personBuilder = ifCanBeParsed { toPersonBuilder(personBuilderProvider.createPersonBuilder(), namespace) } ?: return
-                atomBuilder.addContributorBuilder(personBuilder)
-            }
-            "author" -> {
-                val personBuilder = ifCanBeParsed { toPersonBuilder(personBuilderProvider.createPersonBuilder(), namespace) } ?: return
-                atomBuilder.addAuthorBuilder(personBuilder)
-            }
-            "link" -> {
-                val linkBuilder = ifCanBeParsed { toLinkBuilder(linkBuilderProvider.createLinkBuilder()) } ?: return
-                atomBuilder.addLinkBuilder(linkBuilder)
-            }
+            "contributor" -> ifCanBeParsed { toPersonBuilder(personBuilderProvider.createPersonBuilder(), namespace) }
+                ?.let(atomBuilder::addContributorBuilder)
```

Comment:
Parsers suffered from both MaxLineLength and ReturnCount. Solved with `?.let` and by only passing the function pointer.